torchdata/dataloader2/adapter.py

```diff
@@ -0,0 +1,47 @@
+# Copyright (c) Meta Platforms, Inc. and affiliates.
+# All rights reserved.
+#
+# This source code is licensed under the BSD-style license found in the
+# LICENSE file in the root directory of this source tree.
+
+from abc import abstractmethod
+
+import torch
+
+from torchdata.datapipes.iter import IterDataPipe
+
+__all__ = [
+    "Adapter",
+    "Shuffle",
+]
+
+assert __all__ == sorted(__all__)
+
+
+class Adapter:
+    @abstractmethod
+    def __call__(self, datapipe: IterDataPipe) -> IterDataPipe:
+        pass
+
+
+class Shuffle(Adapter):
+    r"""
+    Shuffle DataPipes adapter allows control over all existing Shuffler (`shuffle`) DataPipes in the graph.
```

Comment:
```suggestion
    Shuffle DataPipes adapter allows control over all existing Shuffler (``shuffle``) DataPipes in the graph.
```
Nit: need double backticks for docstrings (that's how sphinx is)